test/models/rhizome_test.rb

```diff
@@ -21,29 +21,28 @@ def setup
   end
 
   test 'device id should not be too long' do
-    @rhizome.device_id = 'a' * 24
+    @rhizome.particle_device.device_id = 'a' * 34
     assert_not @rhizome.valid?
   end
 
   test 'access token should not be too long' do
-    @rhizome.access_token = 'a' * 40
+    @rhizome.particle_device.access_token = 'a' * 41
     assert_not @rhizome.valid?
   end
 
   test 'device id should not be too short' do
-    @rhizome.device_id = 'a' * 22
+    @rhizome.particle_device.device_id = 'a' * 22
     assert_not @rhizome.valid?
   end
 
   test 'access token should not be too short' do
-    @rhizome.access_token = 'a' * 38
+    @rhizome.particle_device.access_token = 'a' * 38
     assert_not @rhizome.valid?
   end
 
   test 'device ids should be unique' do
-    duplicate_rhizome = @rhizome.dup
-    @rhizome.save
-    assert_not duplicate_rhizome.valid?
+    @rhizome.particle_device.device_id = @second_rhizome.particle_device.device_id
+    assert_not @rhizome.valid?
```

Comment:
Particle model needs uniqueness: true to pass this test. It is enforced at the db level by a unique index, but not at the model level.